k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/response/SearchResponse.java

```diff
@@ -0,0 +1,71 @@
+package com.fsck.k9.mail.store.imap.response;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.fsck.k9.mail.store.imap.ImapResponse;
+import com.fsck.k9.mail.store.imap.Responses;
+import com.fsck.k9.mail.store.imap.command.ImapCommandFactory;
+
+import static com.fsck.k9.mail.store.imap.ImapResponseParser.equalsIgnoreCase;
+
+
+public class SearchResponse extends BaseResponse {
+
+    private final List<Long> numbers;
+
+    private SearchResponse(ImapCommandFactory commandFactory, List<Long> numbers) {
+        super(commandFactory);
+        this.numbers = numbers;
+    }
+
+    public static SearchResponse parseMultiple(ImapCommandFactory commandFactory, List<List<ImapResponse>> responsesList) {
+
+        List<Long> numbers = new ArrayList<>();
+
+        //Currently searching is done only on the basis of either uids or sequence numbers
+        //So it is ok to take the union of all the responses
+        for (List<ImapResponse> responseList : responsesList) {
+            for (ImapResponse response : responseList) {
+                parseSingleLine(response, numbers);
+            }
+        }
+
+        return new SearchResponse(commandFactory, numbers);
+    }
+
+    public static SearchResponse parse(ImapCommandFactory commandFactory, List<ImapResponse> responses) {
+
+        List<Long> numbers = new ArrayList<>();
+
+        for (ImapResponse response : responses) {
+            parseSingleLine(response, numbers);
+        }
+
+        return new SearchResponse(commandFactory, numbers);
+    }
+
+    private static void parseSingleLine(ImapResponse response, List<Long> numbers) {
+        if (response.isTagged() || response.size() < 2 || !equalsIgnoreCase(response.get(0), Responses.SEARCH)) {
```

Comment:
Are there circumstances where this happens and it's not the imap server misbehaving? I feel like this should be a `Timber.w` or even an exception, if we are really getting bogus responses.

(I know this line is not yours, just bringing it up since we're touching it)